parcels/particleset.py

```diff
@@ -249,13 +249,14 @@ def execute(self, pyfunc=AdvectionRK4, starttime=None, endtime=None, dt=1.,
         assert(timeleaps >= 0)
         leaptime = starttime
         for _ in range(timeleaps):
-            leaptime += interval
-            self.kernel.execute(self, endtime=leaptime, dt=dt,
-                                recovery=recovery)
+            # First write output_file, because particles could have been added
```

Comment:
OK, so I've implemented a final write of ParticleFile at then end of .execute now (see c3bd535)